### ISSUE TITLE
TypeError with `dashes`  kwarg to lineplot since 0.12

### ISSUE THREAD

**Issue by matthew-brett** [mwaskom/seaborn#3230]
We just noticed that code that used to work for 0.11, now causes a TypeError.

Here is a mimimal reproducible example:

```python
import numpy as np
import pandas as pd

import seaborn as sns

df = pd.DataFrame(np.random.normal(size=(10, 2)))

sns.lineplot(data=df, dashes=[(5, 5), (10, 10)])
```

This gives the expected plot with Seaborn < 0.12.  For versions 0.12 through current master (72faf27f), it gives:

```
Traceback (most recent call last):
  File "/Volumes/zorg/mb312/tmp/sns_tester.py", line 8, in <module>
    sns.lineplot(data=df, dashes=[(5, 5), (10, 10)])
  File "/Volumes/zorg/mb312/dev_trees/seaborn/seaborn/relational.py", line 645, in lineplot
    p.plot(ax, kwargs)
  File "/Volumes/zorg/mb312/dev_trees/seaborn/seaborn/relational.py", line 459, in plot
    lines = ax.plot(sub_data["x"], sub_data["y"], **kws)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/axes/_axes.py", line 1743, in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/axes/_base.py", line 273, in __call__
    yield from self._plot_args(this, kwargs)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/axes/_base.py", line 418, in _plot_args
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/axes/_base.py", line 418, in <listcomp>
    return [func(x[:, j % ncx], y[:, j % ncy], kw, kwargs)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/axes/_base.py", line 312, in _makeline
    seg = mlines.Line2D(x, y, **kw)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/lines.py", line 390, in __init__
    self.update(kwargs)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/artist.py", line 998, in update
    ret.append(func(v))
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/lines.py", line 1275, in set_dashes
    self.set_linestyle((0, seq))
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/lines.py", line 1141, in set_linestyle
    self._us_dashOffset, self._us_dashSeq = _get_dash_pattern(ls)
  File "/Users/mb312/Library/Python/3.9/lib/python/site-packages/matplotlib/lines.py", line 60, in _get_dash_pattern
    dsum = sum(dashes)
TypeError: unsupported operand type(s) for +: 'int' and 'tuple'
```

This is in a clean virtualenv on an Intel Mac with Homebrew Python 3.9.15 and these modules (`$ pip list`):

```
Package         Version     Editable project location
--------------- ----------- -------------------------------------
contourpy       1.0.7
cycler          0.11.0
fonttools       4.38.0
kiwisolver      1.4.4
matplotlib      3.6.3
numpy           1.24.1
packaging       23.0
pandas          1.5.3
Pillow          9.4.0
pip             22.3.1
pyparsing       3.0.9
python-dateutil 2.8.2
pytz            2022.7.1
seaborn         0.13.0.dev0 /Volumes/zorg/mb312/dev_trees/seaborn
setuptools      65.6.3
six             1.16.0
wheel           0.38.4
```

The setup giving no error differs only in having `seaborn=0.11.2` (and Scipy) instead of current master.

**Comment by mwaskom** [comment]
Likely caused by https://github.com/mwaskom/seaborn/pull/2449/files (based on the release notes the relevant change was folded in there, although it is only tangentially related to the main thrust of that PR).

**Comment by mwaskom** [comment]
Closed by #3316
